conf/etc/nginx/sites-available/default

```diff
@@ -17,16 +17,22 @@
 # Please see /usr/share/doc/nginx-doc/examples/ for more detailed examples.
 ##
 
+# Redirect to HTTPS
 server {
 	listen 80 default_server;
-	listen [::]:80 default_server ipv6only=on;
+	listen [::]:80 default_server;
+	server_name localhost;
+	return 301 https://$server_name$request_uri;
+}
+# HTTPS server
+#
+server {
+	listen 443;
+	server_name localhost ipv6only=on;
```

Comment:
Responses:

- In production, this is set to the IP of an external interface, yes. But I set it to localhost in the repo because any specific IP in production will be different. Setting it to localhost requires that the system hosting it has localhost mapped to the IP of the external interface. Either that's the case, or a deployment step is to replace localhost with the IP being used for that deployment. 
- I added that based on someone's writeup of doing forwarding, not for any conscious reason. Looking at this [stackexchange comment](https://serverfault.com/questions/638367/do-you-need-separate-ipv4-and-ipv6-listen-directives-in-nginx) it seems that you're right, it should be on the listen directive and using it is only relevant to different handling of ipv6 and ipv4. I think this can actually just get removed.